scopes/react/react/apps/web/react-app-options.ts

```diff
@@ -119,4 +120,10 @@ export type ReactAppOptions = {
    * favicon for the app. You can pass an abs path (using require.resolve()) or a url.
    */
   favicon?: string;
+
+  /**
+   * dependencies to be configured for the application - e.g. version of react
+   * items here will override any defaults that are already configured for a package+dependencyType
+   */
+  dependencies?: WorkspacePolicy;
```

Comment:
please move all these dependencies changes into a different PR
I want to review it more carefully, and I'm going to touch the exact same area just now, so we should be in sync on this.
Keeping this inside this PR will block me from merging all the other refactor / renames changes.

Reply:
it's the only way to make the env overridable for the react17/18/etc apps to not have to create a new app-type

I'll move it to another PR but this will block the react18 env...